pkg/cmd/server/start/start_api.go

```diff
@@ -0,0 +1,77 @@
+package start
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+const apiLong = `Start the master API
+
+This command starts the master API.  Running
+
+  $ %[1]s start master %[2]s
+
+will start the server listening for incoming API requests. The server
+will run in the foreground until you terminate the process.`
+
+// NewCommandStartMasterAPI starts only the APIserver
+func NewCommandStartMasterAPI(name, basename string, out io.Writer) (*cobra.Command, *MasterOptions) {
+	options := &MasterOptions{Output: out}
+	options.DefaultsFromName(basename)
+
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Launch master API",
+		Long:  fmt.Sprintf(apiLong, basename, name),
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(); err != nil {
+				fmt.Fprintln(c.Out(), kcmdutil.UsageError(c, err.Error()))
+				return
+			}
+
+			if len(options.ConfigFile) == 0 {
+				fmt.Fprintln(c.Out(), kcmdutil.UsageError(c, "--config is required for this command"))
+				return
+			}
+
+			if err := options.Validate(args); err != nil {
+				fmt.Fprintln(c.Out(), kcmdutil.UsageError(c, err.Error()))
+				return
+			}
+
+			startProfiler()
+
+			if err := options.StartMaster(); err != nil {
+				if kerrors.IsInvalid(err) {
+					if details := err.(*kerrors.StatusError).ErrStatus.Details; details != nil {
+						fmt.Fprintf(c.Out(), "Invalid %s %s\n", details.Kind, details.Name)
+						for _, cause := range details.Causes {
+							fmt.Fprintf(c.Out(), "  %s: %s\n", cause.Field, cause.Message)
+						}
+						os.Exit(255)
+					}
+				}
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	options.MasterArgs = NewDefaultMasterArgs()
+	options.MasterArgs.StartAPI = true
+
+	flags := cmd.Flags()
+	// This command only supports reading from config and the listen argument
+	flags.StringVar(&options.ConfigFile, "config", "", "Location of the master configuration file to run from. Required")
+	cmd.MarkFlagFilename("config", "yaml", "yml")
+	// TODO: add health and metrics endpoints on this listen argument
+	BindListenArg(options.MasterArgs.ListenArg, flags, "")
```

Comment:
I'm confused what you're asking.

On Fri, Aug 14, 2015 at 1:26 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/server/start/start_api.go
> <https://github.com/openshift/origin/pull/4132#discussion_r37098424>:
>
> > +					}
> > +				}
> > +				glog.Fatal(err)
> > +			}
> > +		},
> > +	}
> > +
> > +	options.MasterArgs = NewDefaultMasterArgs()
> > +	options.MasterArgs.StartAPI = true
> > +
> > +	flags := cmd.Flags()
> > +	// This command only supports reading from config and the listen argument
> > +	flags.StringVar(&options.ConfigFile, "config", "", "Location of the master configuration file to run from. Required")
> > +	cmd.MarkFlagFilename("config", "yaml", "yml")
> > +	// TODO: add health and metrics endpoints on this listen argument
> > +	BindListenArg(options.MasterArgs.ListenArg, flags, "")
>
> can we at least let the API use the configured bind address, and just let
> the controllers command do the weird config+arg thing? controllers would
> *just* expose health/metrics... if you change the API port here, you
> probably invalidate a lot of other bits of the config
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4132/files#r37098424>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift